app/javascript/components/users/SignInPage.spec.tsx

```diff
@@ -70,3 +70,14 @@ describe('SignInPage', () => {
 		waitFor(() => expect(error).toHaveTextContent("Ermahgerd! We had an error!"));
 	});
 });
+
+//Testing error boundary
+describe('<SignInPage>', () => {
+	it('error boundary', () => {
+		function Fallback() {
+			return <div>This is an error</div>
+		}
+	});
+	const error = 'Oh no!';
+	expect(error).toHaveTextContent("This is an error");
+});
```

Comment:
I think there's a misunderstanding of how jest tests work. A test goes inside an `it` call. Assertions are tested using  `expect` function but those go inside a test. So your rough structure of this file should be should be:
```javascript

// put any helper components here

describe('SignInPage', () => {
  it('has an error', async () => { // the name "has an error' summarizes what this test asserts.
          expect.assertions(1) // this is the number of assertions that will happen in this test
         const result =  render(<SignInPage> </SignInPage) // render whatever needs to be done to make the error throw
        const error = await result.findByTestId('error'); // this assumes the test id for the error div is "error"
   	expect(error).toHaveTextContent("This is an error"); // everytime you call expect is an assertion
        // you could have more assertions here, if you need to check additional things to prove that what you're checking is true
  });

 it('has ... ', async () => {
   // set up the situation and make some assertions to prove what is being tested in this test in true
 });
 // any more tests
})

Reply:
Thank you, Eric! We are currently working on these tests. We've forgot to add a comment saying that this is a work in progress.